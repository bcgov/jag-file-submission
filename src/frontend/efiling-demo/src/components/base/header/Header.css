.navbar > .container-fluid {
  width: 95%;
  justify-content: inherit;
}

.pointer {
  cursor: pointer;
}
