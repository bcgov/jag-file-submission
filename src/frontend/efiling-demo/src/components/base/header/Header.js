/* eslint-disable no-alert */
import React from "react";
import PropTypes from "prop-types";
import { useHistory } from "react-router-dom";
import "./Header.css";

export const HeadingTitle = (history, classNames) => (
  <div
    className={classNames}
    onClick={() => history.push("/")}
    role="button"
    onKeyDown={() => history.push("/")}
    tabIndex={0}
    aria-labelledby="title"
  />
);

export const HeaderImage = (classNames, width, src) => (
  <img
    className={classNames}
    src={`${process.env.PUBLIC_URL}/images/${src}`}
    width={width}
    height="44"
    alt="B.C. Government Logo"
  />
);

export default function Header({ header: { name } }) {
  const history = useHistory();

  return (
    <header>
      <nav className="container-fluid navbar navbar-expand-lg navbar-dark">
        {HeadingTitle(history, "navbar-brand pointer")}
        {HeaderImage(
          "img-fluid d-none d-md-block",
          181,
          "bcid-logo-rev-en.svg"
        )}
        {HeaderImage("img-fluid d-md-none", 64, "bcid-symbol-rev.svg")}
        {HeadingTitle(history, "pointer navbar-brand nav-item nav-link")}
        <div id="title" className="navbar-brand">
          {name}
        </div>
      </nav>
    </header>
  );
}

Header.propTypes = {
  header: PropTypes.shape({
    name: PropTypes.string.isRequired
  }).isRequired
};
