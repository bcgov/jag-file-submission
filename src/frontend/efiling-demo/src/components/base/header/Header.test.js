import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import testBasicSnapshot from "../../../TestHelper";

import Header, { HeadingTitle, HeaderImage } from "./Header";

describe("Header Component", () => {
  const header = {
    name: "File Submission"
  };

  test("Header matches the snapshot", () => {
    const history = createMemoryHistory();

    const headerComponent = (
      <Router history={history}>
        <Header header={header} />
      </Router>
    );

    testBasicSnapshot(headerComponent);
  });

  test("HeadingTitle matches the snapshot", () => {
    const history = createMemoryHistory();

    const headingTitle = HeadingTitle(history, "navbar-brand pointer");

    testBasicSnapshot(headingTitle);
  });

  test("HeaderImage matches the snapshot", () => {
    const headingImage = HeaderImage(
      "img-fluid d-none d-md-block",
      181,
      "bcid-logo-rev-en.svg"
    );

    testBasicSnapshot(headingImage);
  });
});
