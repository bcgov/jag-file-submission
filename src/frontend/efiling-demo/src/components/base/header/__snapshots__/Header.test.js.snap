// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header matches the snapshot 1`] = `
<header>
  <nav
    className="container-fluid navbar navbar-expand-lg navbar-dark"
  >
    <div
      aria-labelledby="title"
      className="navbar-brand pointer"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    />
    <img
      alt="B.C. Government Logo"
      className="img-fluid d-none d-md-block"
      height="44"
      src="undefined/images/bcid-logo-rev-en.svg"
      width={181}
    />
    <img
      alt="B.C. Government Logo"
      className="img-fluid d-md-none"
      height="44"
      src="undefined/images/bcid-symbol-rev.svg"
      width={64}
    />
    <div
      aria-labelledby="title"
      className="pointer navbar-brand nav-item nav-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    />
    <div
      className="navbar-brand"
      id="title"
    >
      File Submission
    </div>
  </nav>
</header>
`;

exports[`Header Component HeaderImage matches the snapshot 1`] = `
<img
  alt="B.C. Government Logo"
  className="img-fluid d-none d-md-block"
  height="44"
  src="undefined/images/bcid-logo-rev-en.svg"
  width={181}
/>
`;

exports[`Header Component HeadingTitle matches the snapshot 1`] = `
<div
  aria-labelledby="title"
  className="navbar-brand pointer"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
/>
`;
